gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -120,6 +123,7 @@ public abstract class KafkaJobStatusMonitor extends HighLevelConsumer<byte[], by
   private final GaaSJobObservabilityEventProducer eventProducer;
   private final DagManagementStateStore dagManagementStateStore;
   private final boolean dagProcEngineEnabled;
+  private final List<Class<? extends Exception>> nonTransientExceptions = Collections.singletonList(SQLIntegrityConstraintViolationException.class);
```

Comment:
I wouldn't name these "non-transient" exceptions, so much as "permitted"/"allowable" exceptions

Reply:
how about nonRetryableExceptions ? because that's the purpose this list is serving - deciding to retry or not. permitted/allowed gives a sense like the exceptions are not something to worry about at all